src/helpers/ht_data.service.ts

```diff
@@ -14,13 +14,24 @@ export class HT_DataService {
     private session_key: string;
     private local_key: string;
 
-    constructor(username: string) {
-        username = username.toLowerCase();
-        this.setUsername(username);
-        let data = this.getData();
-        if (!data.saltBits) {
-            data.saltBits = JSON.stringify(HaventecCommon.generateSalt());
-            this.setData(data);
+    constructor(username?: string) {
+
+        if ( !username ) {
+            const haventec_username = HT_LocalStorage.getItem('haventec_username');
+            if ( haventec_username ) {
+                username = haventec_username.toString();
+            }
+        }
+
+        if ( username ) {
+            username = username.toLowerCase().replace(/\"/g, '');
+
+            this.setUsername(username);
+            let data = this.getData();
```

Comment:
Need typing here, hard to infer by reading this

Reply:
Set to type HT_Data